### ISSUE TITLE
Angular i18n error for translation file merging and zh-CN

### ISSUE THREAD

**Issue by xiaoxiangmoe** [angular/angular-cli#18301]
# 🐞 Bug report

### Command (mark with an `x`)

- [ ] new
- [x] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [x] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Is this a regression?

 Yes, the previous version in which this bug was not present was: ....

No

### Description

 A clear and concise description of the problem...

## 🔬 Minimal Reproduction

https://github.com/xiaoxiangmoe/angular-i18n-error

## 🔥 Exception or Error

```json
      "i18n": {
        "sourceLocale": "en-US",
        "locales": {
          "zh-CN": {
            "translation": ["src/locale/messages.zh-CN.xlf"]
          }
        }
      }
```
<pre><code>An unhandled exception occurred: Workspace config file cannot be loaded: /Users/zhaojinxiang/repos/play-angular-lib/angular.json
Schema validation failed with the following errors:
  Data path ".projects['play-angular-lib'].i18n.locales['zh-CN']" should be string.
  Data path ".projects['play-angular-lib'].i18n.locales['zh-CN'].translation" should be string.
  Data path ".projects['play-angular-lib'].i18n.locales['zh-CN']" should match exactly one schema in oneOf.
See "/private/var/folders/b3/pwyf75ts5ks08jqfxmhzbv1r0000gq/T/ng-sxwHMO/angular-errors.log" for further details.</code></pre>

if we change i18n config into 

```json
      "i18n": {
        "sourceLocale": "en-US",
        "locales": {
          "zh-CN": {
            "translation": "src/locale/messages.zh-CN.xlf"
          }
        }
      }
```
<pre><code>Locale data for 'zh-CN' cannot be found.  Using locale data for 'zh'.</code></pre>

if we change i18n config into 

```json
      "i18n": {
        "sourceLocale": "en-US",
        "locales": {
          "zh": {
            "translation": "src/locale/messages.zh-CN.xlf"
          }
        }
      }
```

it works well

## 🌍 Your Environment
<pre><code>Angular CLI: 10.1.0-next.1
Node: 14.5.0
OS: darwin x64

Angular: 10.1.0-next.1
... animations, cli, common, compiler, compiler-cli, core, forms
... localize, platform-browser, platform-browser-dynamic, router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.1001.0-next.1
@angular-devkit/build-angular     0.1001.0-next.1
@angular-devkit/build-optimizer   0.1001.0-next.1
@angular-devkit/build-webpack     0.1001.0-next.1
@angular-devkit/core              10.1.0-next.1
@angular-devkit/schematics        10.1.0-next.1
@ngtools/webpack                  10.1.0-next.1
@schematics/angular               10.1.0-next.1
@schematics/update                0.1001.0-next.1
rxjs                              6.6.0
typescript                        3.9.7
webpack                           4.43.0</code></pre>

**Anything else relevant?**

https://github.com/angular/angular-cli/issues/18276

https://github.com/angular/angular-cli/issues/17126

**Comment by xiaoxiangmoe** [comment]
@petebacondarwin

**Comment by petebacondarwin** [comment]
Hi @xiaoxiangmoe  thanks for the issue.

Translation file merging is not yet implemented in the CLI - see https://github.com/angular/angular-cli/issues/18276

The second warning `Locale data for 'zh-CN' cannot be found.  Using locale data for 'zh'.` is working as expected. See https://github.com/angular/angular-cli/pull/17145

Closing as a duplicate of #18276

**Comment by xiaoxiangmoe** [comment]
@petebacondarwin when will angular-cli 10.1 release (with translation file merging)?

**Comment by petebacondarwin** [comment]
@clydin or @kyliau? ^^^

**Comment by kyliau** [comment]
This will be part of the v10.1.0 release scheduled for August 26, 2020.
Prior to that, it'll be in v10.1.0-rc.0 scheduled for August 19, 2020.

**Comment by xiaoxiangmoe** [comment]
@kyliau TS 4.0 will release at  August 20, 2020. Will it included in v10.1.0 release at August 26, 2020?

**Comment by alan-agius4** [comment]
@xiaoxiangmoe, TS 4.0 support is targeted to be added in Angular 10.2.0

**Comment by kyliau** [comment]
Yes, Angular 10.2.0 is scheduled for October 7, 2020.

**Comment by xiaoxiangmoe** [comment]
@alan-agius4 

I'm using `"typescript": "~4.1.0-dev.20200820"` in Angular 10.0:

```
Error: Error on worker #1: TypeError: ts.updateIdentifier is not a function
```

Can we fix it in angular 10.1 or angular 10.2.0-alpha ? I need update my project to test new TypeScript feature https://github.com/microsoft/TypeScript/pull/39669

**Comment by petebacondarwin** [comment]
@xiaoxiangmoe - see https://github.com/angular/angular/pull/38543
